apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -113,51 +109,54 @@
      */
     handleEvent: function(e) {
 
+      var self = this;
+
       function cycleElements(index) {
 
-        var removed = this.elements.splice(-index, index);
+        var removed = self.elements.splice(-index, index);
 
         for (var i = 0; i < removed.length; i++) {
 
-          this.elemList.insertBefore(removed[i], this.elements[0 + i]);
-          this.elements.splice(0 + i, 0, removed[i]);
+          self.elemList.insertBefore(removed[i], self.elements[0 + i]);
```

Comment:
why "0 + i" instead of simple i? Am I missing something here? :) and also on line 121